lib/Doctrine/DBAL/DriverManager.php

```diff
@@ -220,14 +218,7 @@ public static function getConnection(
         }
 
         if (isset($params['pdo'])) {
-            Deprecation::trigger(
-                'doctrine/dbal',
-                'https://github.com/doctrine/dbal/pull/3554',
-                'Passing a user provided PDO instance directly to Doctrine is deprecated.'
-            );
-
             $params['pdo']->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
```

Comment:
Seems a bit risky to me. This will change the current behavior by overriding the error mode for someone providing a 'pdo' params without using DriverManager.

```php
$pdo = new PDO('sqlite::memory:');
$pdo->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_SILENT);

$conn = new \Doctrine\DBAL\Connection(['pdo' => $pdo], $driver);
```